pkg/ebpf/events_pipeline.go

```diff
@@ -256,15 +256,23 @@ func (t *Tracee) decodeEvents(ctx context.Context, sourceChan chan []byte) (<-ch
 			evt.ProcessEntityId = utils.HashTaskID(eCtx.HostPid, eCtx.LeaderStartTime)
 			evt.ParentEntityId = utils.HashTaskID(eCtx.HostPpid, eCtx.ParentStartTime)
 
+			policies, err := policy.Snapshots().Get(evt.PoliciesVersion)
+			if err != nil {
+				t.handleError(err)
+				t.eventsPool.Put(evt)
+				continue
+			}
+
 			// If there aren't any policies that need filtering in userland, tracee **may** skip
 			// this event, as long as there aren't any derivatives or signatures that depend on it.
 			// Some base events (derivative and signatures) might not have set related policy bit,
 			// thus the need to continue with those within the pipeline.
-			if t.matchPolicies(evt) == 0 {
+			if t.matchPolicies(policies, evt) == 0 {
+				// TODO: as isReqBySignature, eventDerivations must also be moved to policy package
 				_, hasDerivation := t.eventDerivations[eventId]
-				_, hasSignature := t.eventSignatures[eventId]
+				isReqBySignature := policies.EventsFlags().Get(events.ID(evt.EventID)).RequiredBySignature()
```

Comment:
You are right. And after delving into this once again, I now consider that `EventFlags` should hold the static properties of an event, where, for example, the `RequiredBySignature` information should be, and it should reside in the events pkg.

On the other hand `EventStates` should hold the computed ones and be managed from and reside in policy package.

WDYT?

Reply:
The relation that we currently have as part of the event definition (which is static) is that some event (e.g. "a signature) may depend on some other event. So this part is already handled by the existing event dependencies but here we want the reverse relation - check if the event is required by some other event (signature), which is of course also a static property. This property can be computed at the initialization of tracee and kept for the whole run since it is static.

And like you said, EventState should hold dynamic data about the event, which can be changed by the policy.